Fix value not being properly tracked in `v-collection-field-template`

ComfortablyCoding opened the pull request:
<!--

Heya! Thanks for opening a Pull Request! If your PR is implementing a new feature or fix for Directus, please make sure your PR adheres to the following requirements:

- The PR closes an Issue (not Discussion)
- Tests are added/updated and are passing locally if applicable
- Documentation was added/updated if applicable

Please make sure to "Link" the issue you're closing. Without a Linked issue, this PR won't be accepted. See https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue for more information.

-->

## Scope

What's changed:

- We now properly track and emit model value changes for this component.
- Removes the warning in the console for missing required prop "inject".
```
Vue warn]: Missing required prop: "inject" 
  at <VCollectionFieldTemplate modelValue="{{id}}" onUpdate:modelValue=fn collection="article"  ... > 
  at <Options collection="article" .....
```

### Sample side effect

Before

https://github.com/directus/directus/assets/44623501/5bc11204-2d29-4f61-b832-4f0a3115135c

After

https://github.com/directus/directus/assets/44623501/07211a37-2a04-42d2-ba9a-4632e7ab3398


## Potential Risks / Drawbacks

- N/A

## Review Notes / Questions

- N/A

---

Fixes N/A, this is a regression from #21174

ComfortablyCoding committed:
fix placeholder not being removed

paescuj commented on app/src/components/v-collection-field-template.vue:
```diff
@@ -4,20 +4,21 @@ import type { Field, Relation } from '@directus/types';
+		placeholder?: string | null;
+		disabled?: boolean;
+		collection: string | null;
+		inject?: {
+			fields: Field[];
+			relations?: Relation[];
+		} | null;
+	}>(),
+	{ inject: null },
```
I think we could omit the default value, disallow `null` and just make it optional.
FWIW, the `inject` option isn't used anywhere at the moment - we might also remove it from this component and add it back if there's demand for it.

paescuj commented on app/src/components/v-collection-field-template.vue:
```diff
@@ -4,20 +4,21 @@ import type { Field, Relation } from '@directus/types';
-}>();
+const props = withDefaults(
+	defineProps<{
+		modelValue?: string | null;
```
Let's use the new `defineModel` macro: https://vuejs.org/guide/components/v-model.html

ComfortablyCoding replied:
Done in https://github.com/directus/directus/pull/21855/commits/c32d3742f037fdec4e05afe00705bf21463c1abb

ComfortablyCoding committed:
remove inject and utilize defineModel macro

paescuj committed:
Move defineModel before defineProps

Bit of personal preference for now, since we just started using defineModel but
I think we should try to maintain this order

paescuj reviewed (approved):
❤️
